### PR TITLE
.github: Add pull request template

### PR DESCRIPTION
**Pull request by samuelkarp**
### Summary
Add a new pull request template to help standardize the information we want as part of a pull request, inspired by the template used by the Docker project.

### Implementation details

A markdown file at `.github/PULL_REQUEST_TEMPLATE.md`. See [the documentation](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/).

### Testing

- [x] Builds (`make release`)
- [x] Unit tests (`make short-test`) pass
- [ ] Integration tests (`make test`) pass
- [ ] Functional tests (`make run-functional-tests`) pass

New tests cover the changes: no

### Description for the changelog

No changelog entry

### Licensing

This contribution is under the terms of the Apache 2.0 License: yes (Amazon employee)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Unit tests (`make short-test`) pass
+- [ ] Integration tests (`make test`) pass
+- [ ] Functional tests (`make run-functional-tests`) pass
+
+New tests cover the changes: <!-- yes|no -->
+
+### Description for the changelog
+<!--
+Write a short (one line) summary that describes the changes in this
+pull request for inclusion in the changelog:
```
Comment (aaithal) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we add a link here to an existing entry in the changelog that can serve as a reference?

**Comment by aaithal** [comment]
Nice! Thanks for doing this. I have a suggestion in the comment. Let me know what you think about that.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (samuelkarp) [inline .github/PULL_REQUEST_TEMPLATE.md]: Sure, any particular one we'd want to use?

Comment (aaithal) [inline .github/PULL_REQUEST_TEMPLATE.md]: I like this [commit](https://github.com/aws/amazon-ecs-agent/commit/c9aefebc2b3007f09468f651f6308136bd7b384f).

**Comment by samuelkarp** [comment]
@aaithal Updated

**Comment by aaithal** [comment]
:shipit:

**Pull request merged**